frontend/src/scenes/data-warehouse/settings/source/SourceConfiguration.tsx

```diff
@@ -0,0 +1,58 @@
+import { LemonBanner, LemonButton, LemonSkeleton } from '@posthog/lemon-ui'
+import { BindLogic, useValues } from 'kea'
+import { Form } from 'kea-forms'
+import { SourceFormComponent, SourceFormProps } from 'scenes/data-warehouse/external/forms/SourceForm'
+
+import { dataWarehouseSourceSettingsLogic } from './dataWarehouseSourceSettingsLogic'
+
+interface SourceConfigurationProps {
+    id: string
+}
+
+export const SourceConfiguration = ({ id }: SourceConfigurationProps): JSX.Element => {
+    const { sourceFieldConfig, sourceLoading } = useValues(dataWarehouseSourceSettingsLogic({ id }))
+    return (
+        <BindLogic logic={dataWarehouseSourceSettingsLogic} props={{ id }}>
+            {sourceLoading && sourceFieldConfig === null ? (
+                <LemonSkeleton />
+            ) : (
+                <UpdateSourceConnectionFormContainer id={id} sourceConfig={sourceFieldConfig} showPrefix={false} />
+            )}
+        </BindLogic>
+    )
+}
+
+interface UpdateSourceConnectionFormContainerProps extends SourceFormProps {
+    id: string
+}
+
+function UpdateSourceConnectionFormContainer(props: UpdateSourceConnectionFormContainerProps): JSX.Element {
+    const { source, sourceLoading } = useValues(dataWarehouseSourceSettingsLogic({ id: props.id }))
+
+    if (source?.source_type !== 'MSSQL' && source?.source_type !== 'MySQL' && source?.source_type !== 'Postgres') {
+        return (
+            <LemonBanner type="warning" className="mt-2">
+                <p>
+                    Only Postgres, MSSQL, and MySQL are configurable. Please delete and recreate your source if you need
+                    to connect to a new source of the same type.
+                </p>
+            </LemonBanner>
+        )
+    }
```

Comment:
Is this because there's not a 1:1 match between the field names and job inputs for the other sources right now?

Reply:
yeah :/